### DISCUSSION TITLE
[教程]如何开通开放平台

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2757] [Developer]
# 重要提示

> ### 嗨，简悦全新的 [教程中心](https://www.yuque.com/kenshin/simpread) 已上线，**下面的内容并不一定是最新的**，建议 [访问这里](https://www.yuque.com/kenshin/simpread/ieu1w6) 查看最新版教程。

***

### 打开选项页

> 以下全部的内容都在选项页。

![](https://cdn.nlark.com/yuque/0/2022/png/353945/1668230896613-5db25b58-537d-41d5-a1cd-2c8082127df4.png#averageHue=%23f0f0ef&clientId=u4ec906c6-7357-4&crop=0&crop=0&crop=1&crop=1&id=jGgQw&originHeight=263&originWidth=483&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&taskId=uf05d862d-3d6b-49ca-b240-b785065f292&title=)
![](https://cdn.nlark.com/yuque/0/2022/png/353945/1668230896845-31fbe83a-a59c-4a8f-a939-2aebfc6875b2.png#averageHue=%2373765a&clientId=u4ec906c6-7357-4&crop=0&crop=0&crop=1&crop=1&id=A2RY5&originHeight=434&originWidth=486&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&taskId=uc945a200-0eeb-40f1-a5b9-49cf9963fd6&title=)

### 授权登录

> **简悦不托管用户的数据，用户数据必须在用户手中，这样才能保障用户拥有真正意义的数据所有权。为了更好的让用户使用 API，所以需要用户将坚果云「托管」给简悦使用。**
> **简悦 不会明文保存你的用户名/密码，请放心使用，因为采用了 非对称加密 方式，所以除了你没有人知道你的密码。**

> **下面是开通方法。**

#### 切换到坚果云

> 选项页 → 共通 → 同步盘，切换到坚果云。

![SCR-20221112-ikc.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668230555949-6ccbf115-92ce-4504-acef-b6b0cd16d1cf.png#averageHue=%23f7f7f7&clientId=u4f4da5e0-8602-4&crop=0&crop=0&crop=1&crop=1&from=ui&id=u16376f06&name=SCR-20221112-ikc.png&originHeight=223&originWidth=876&originalType=binary&ratio=1&rotation=0&showTitle=false&size=51663&status=done&style=none&taskId=u9f041e68-bbaf-4797-ab2b-3d233a848e0&title=)

#### 授权坚果云

> 打开选项页 → 服务 → 授权管理，授权坚果云。

> 如何获取坚果云的话，[请看这里](http://ksria.com/simpread/docs/#/%E5%9D%9A%E6%9E%9C%E4%BA%91) 。

![](https://z3.ax1x.com/2021/05/04/gnQynK.png#crop=0&crop=0&crop=1&crop=1&id=R7DFm&originHeight=481&originWidth=1249&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&title=)

#### 授权坚果云给简悦

> 打开 **选项页 → 右上角 → 使用第三方作为登录**，并按下图所示操作。

![2022-11-12_13-15-05 (1).gif](https://cdn.nlark.com/yuque/0/2022/gif/353945/1668230942066-3bb612af-d959-496e-9cf8-6a5a49528d81.gif#averageHue=%2386adec&clientId=u4ec906c6-7357-4&crop=0&crop=0&crop=1&crop=1&from=ui&id=ue9ba8680&name=2022-11-12_13-15-05%20%281%29.gif&originHeight=892&originWidth=1846&originalType=binary&ratio=1&rotation=0&showTitle=false&size=1505101&status=done&style=none&taskId=ub236ea59-c373-4ee1-bcd2-2fa6d82330a&title=)

> 注意：如果上图显示不是坚果云而是 Dropbox 请在附录中查看解决方案。

### 同步配置文件

当配置完上面后，一定 **确保你的坚果云存在**  `simpread_config.json` 文件，所以需要使用 [同步功能](http://ksria.com/simpread/docs/#/%E5%90%8C%E6%AD%A5)，可以使用下面两种方案。

![](https://z3.ax1x.com/2021/05/04/gn3JnP.png#crop=0&crop=0&crop=1&crop=1&id=P3iWx&originHeight=436&originWidth=1576&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&title=)

#### 手动同步

如果你只是轻量级使用简悦，且没有使用 [同步助手](http://ksria.com/simpread/docs/#/Sync) 的话，请使用 [手动同步](http://ksria.com/simpread/docs/#/%E5%90%8C%E6%AD%A5?id=%e6%89%8b%e5%8a%a8%e5%90%8c%e6%ad%a5) 方案。
如果是 **首次使用这些的话，一定要先同步一次** 配置文件。

#### 自动同步

自动同步方案 **依赖坚果云客户端**，也就是说需要保证：[同步助手](http://ksria.com/simpread/docs/#/Sync) 与 坚果云 同时运行即可，如何配置请看 [自动同步](http://ksria.com/simpread/docs/#/%E8%87%AA%E5%8A%A8%E5%90%8C%E6%AD%A5) 方案。

如果提示保存成功，在本地的稍后读没有此 URL 时，请查看坚果云 `simpread_config.json` 的日期是否更新了。

### 登录开放平台

> 地址 [https://simpread.pro/developer](https://simpread.pro/developer)

> 可使用 **UID & License / ID & Short /  坚果云 / 免密码  登录。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668231041211-d7485ce2-71db-41d9-8df6-e013fc126fd4.png#averageHue=%23f9f9f9&clientId=u4ec906c6-7357-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=892&id=uf47b899a&name=image.png&originHeight=1784&originWidth=3690&originalType=binary&ratio=1&rotation=0&showTitle=false&size=289173&status=done&style=none&taskId=u3637cb27-5c08-4b8f-93e2-6d3d12b2db1&title=&width=1845)

#### 注意

1. 确保在扩展端已经使用 授权登录。
2. **首次登录时，请使用 UID / License 登录方案**。

#### UID 登录方案

> 首次登录时，请使用此方式，这是最简单的方案。
> 如果不知道 UID 可通过在附录中找到解决方案。

#### Short 登录方案

> 当登录到开放平台后，可以设置用户名和**密码（在这里叫做 Short 授权码） **登录，细节请看下面的说明。

#### 坚果云登录

> 当开通 **授权登录** 并在坚果云中存在 `simpread_config.json` 后也可以使用坚果云的账户 Email / WebDAV 密码 登录。

### 功能介绍

> 登录到开放平台后，介绍几个经常使用的功能。

#### 设置 Token

> 当登录成功后，简悦会自动分配当前用户一个 Token，当获取后，就可以使用简悦 API 提供的 [各种工具](https://github.com/Kenshin/simpread/discussions/tools) 了。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668228793567-51d9d11b-35e1-4cc7-8819-0ea48f3d4a1d.png#averageHue=%23f1efef&clientId=u5e58894a-0078-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=169&id=ua13627e8&name=image.png&originHeight=338&originWidth=2250&originalType=binary&ratio=1&rotation=0&showTitle=false&size=116900&status=done&style=none&taskId=u53db09d9-045d-4217-a31e-0717f4d7e72&title=&width=1125)

#### 设置 Short

> 登录之后，可以通过设置 ID 与 Short（密码），之后可以使用 Short 方式登录。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668228852780-52c3767f-b141-4727-86a5-e9767f571a8f.png#averageHue=%23f2efee&clientId=u5e58894a-0078-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=319&id=u0aaa8feb&name=image.png&originHeight=638&originWidth=2246&originalType=binary&ratio=1&rotation=0&showTitle=false&size=229620&status=done&style=none&taskId=u419feef9-5275-45e6-898a-817d59b1dc3&title=&width=1123)

#### API 快照

> 在通过 API 收集时，可以将这个 URL 对应的 HTML 也保存到你的坚果云。（暂还不支持图片的保存）
> 保存后的内容，可以在 [阅读列表](https://github.com/Kenshin/simpread/discussions/2954) / 稍后读直接查看。

正如我对隐私权的重视，此功能默认为关闭状态，只有用户允许的话，才可以使用。

![](https://user-images.githubusercontent.com/81074/139578111-f017e4f7-cd41-4e6d-b02a-6b63cadce0bd.png#crop=0&crop=0&crop=1&crop=1&id=O4XQX&originHeight=355&originWidth=720&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&title=)

> 注意：并不是每种 API 工具均支持快照功能，目前包括：

- [Telegram bot](https://kb.simpread.pro/#/page/61f63f40-e012-4800-b29c-7164dd632457)

- [收藏助手](https://github.com/Kenshin/simpread/discussions/2921)

- [直接调用 API 的联动](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3Aapi)

#### 开通其它权限

> **确保下面的选项都开启状态**，尤其是使用 API、收藏助手 等功能时，建议进入开放平台后就启用这些权限。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1665732551149-2bff0e64-2fa5-4fbe-bfbf-347bcb22369c.png#averageHue=%23f7f7f6&clientId=u547eaa96-04be-4&crop=0&crop=0&crop=1&crop=1&from=paste&height=736&id=u5a65ba15&name=image.png&originHeight=1472&originWidth=1902&originalType=binary&ratio=1&rotation=0&showTitle=false&size=571377&status=done&style=none&taskId=ucefea71b-79bc-44be-8715-ab5deb1de9c&title=&width=951)

#### 如何找到 UID

> 打开选项页 → 账户

![](https://cdn.nlark.com/yuque/0/2022/png/353945/1668228224989-bf2cacbc-7757-4129-bf7b-efb32f716710.png#averageHue=%23f6e5e3&clientId=u46a51fb9-2709-4&crop=0&crop=0&crop=1&crop=1&id=abuYn&originHeight=554&originWidth=1097&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&taskId=ubbf88648-d124-4b73-a18a-d4f5dda68b2&title=)

#### 如何找到 License

> 详细说明 [在哪里可以重新找到 License（兑换码）](https://github.com/Kenshin/simpread/discussions/2756)

- 少数派付款渠道 → **登录少数派后台** 或 **检索你的邮件**

- 爱发电渠道 → 登录 **爱发电** → 右上角 **消息**

- 支付宝直接购买渠道 → 登录 [https://simpread.pro/code](https://simpread.pro/code) 输入支付宝订单号后六位

#### 切换为坚果云

> 如果上面显示的内容为下图。

![SCR-20221112-ilb.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668230608448-1e7b1338-ad11-4ebb-9973-ffe7ee985aee.png#averageHue=%23f5eae8&clientId=u4f4da5e0-8602-4&crop=0&crop=0&crop=1&crop=1&from=ui&id=u56fbb595&name=SCR-20221112-ilb.png&originHeight=346&originWidth=871&originalType=binary&ratio=1&rotation=0&showTitle=false&size=64715&status=done&style=none&taskId=u747ca367-80dc-47ec-ba87-a06054ab51a&title=)

> 选项页 → 账户 → 授权管理，请按照下图设置进行切换操作，**切换后请刷新当前页面**。

![SCR-20221112-ilv.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1668230651901-023a12a9-a7e7-4aa0-b841-12b8690685da.png#averageHue=%23f8ebeb&clientId=u4f4da5e0-8602-4&crop=0&crop=0&crop=1&crop=1&from=ui&id=u4c4e8e13&name=SCR-20221112-ilv.png&originHeight=340&originWidth=871&originalType=binary&ratio=1&rotation=0&showTitle=false&size=66716&status=done&style=none&taskId=u401b7e60-fb3d-4118-a06b-439d74cd4f4&title=)

#### 

#### 发生未知错误

> 如果登录 [阅读列表](https://github.com/Kenshin/simpread/discussions/2954) 出现下图截图错误（前提是你使用正确的方式登录）

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1657179343474-bdfd3993-0bdd-4981-b50a-147267f0486a.png#averageHue=%23fefefe&clientId=u14d8f4f1-b463-4&crop=0&crop=0&crop=1&crop=1&errorMessage=unknown%20error&from=paste&height=285&id=ub766e4b6&name=image.png&originHeight=569&originWidth=1280&originalType=binary&ratio=1&rotation=0&showTitle=false&size=33906&status=error&style=none&taskId=u7e02df22-1d93-4572-8a4d-6d99731b143&title=&width=640)

> 是因为没有开启相关权限，需要重新登录 [https://simpread.pro/developer](https://simpread.pro/developer) 后，开启下图权限（选择使用密码登录），并保存。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1657179425298-cc87923b-9891-4c0f-9a6d-45d6cb98f690.png#averageHue=%23f7f7f7&clientId=u14d8f4f1-b463-4&crop=0&crop=0&crop=1&crop=1&errorMessage=unknown%20error&from=paste&height=354&id=ue00563de&name=image.png&originHeight=707&originWidth=1280&originalType=binary&ratio=1&rotation=0&showTitle=false&size=134899&status=error&style=none&taskId=u375e977e-cf7d-4221-ab6d-f4013e334e9&title=&width=640)

#### output 文件夹不存在

> 如果登录 [阅读列表](https://github.com/Kenshin/simpread/discussions/2954) 出现下图错误，说明你的坚果云没有 `output` 文件夹或者 `output` 文件夹没有任何 HTML 文件，需要配置知识库，可以看下 [一站式知识库配置](https://www.yuque.com/kenshin/simpread/wkswh7?view=doc_embed) 。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1657179562008-c0cd09e8-ef85-4f32-9ab0-473ce262e25c.png#averageHue=%23fafafa&clientId=u14d8f4f1-b463-4&crop=0&crop=0&crop=1&crop=1&errorMessage=unknown%20error&from=paste&height=73&id=u447e8f02&name=image.png&originHeight=146&originWidth=744&originalType=binary&ratio=1&rotation=0&showTitle=false&size=21979&status=error&style=none&taskId=u508dfc2b-0d8b-4d8a-8bb4-698ac2fab7c&title=&width=372)

#### 反复登录的问题

> 如果因为不知道登录信息，反复登录会出现下图（或类似的错误，如：页面刷新后无反应等）
> **请在隐私模式下后重新登录**，此问题过一阵子会自动消失。

![image.png](https://cdn.nlark.com/yuque/0/2022/png/353945/1657179875821-b7a6c466-1e1d-4fe5-9a69-2aadbe8260ea.png#averageHue=%23ddd0cb&clientId=u14d8f4f1-b463-4&crop=0&crop=0&crop=1&crop=1&errorMessage=unknown%20error&from=paste&height=172&id=u6f3a3ed6&name=image.png&originHeight=344&originWidth=1126&originalType=binary&ratio=1&rotation=0&showTitle=false&size=244296&status=error&style=none&taskId=uba5e0443-a231-48e1-9673-357098871cf&title=&width=563)
#### 

#### 请确保你的网络畅通

> 当出现下图所示的提示时，**请使用坚果云登录方案。**

![](https://cdn.nlark.com/yuque/0/2022/png/353945/1668228407274-4755cd5f-e5ac-4928-9c25-f31daeb16cc9.png#averageHue=%23bbb2a3&clientId=u5e58894a-0078-4&crop=0&crop=0&crop=1&crop=1&id=DvEAx&originHeight=236&originWidth=505&originalType=binary&ratio=1&rotation=0&showTitle=false&status=done&style=none&taskId=u2eecf54e-6d3a-43c2-a099-0813b434ebc&title=)

### 引申

> 更多开放平台 [相关问题](https://github.com/Kenshin/simpread/discussions/categories/developer?discussions_q=category%3ADeveloper+label%3Awarning)

### 关联阅读

- [配置文件](http://ksria.com/simpread/docs/#/配置文件)

- [坚果云](http://ksria.com/simpread/docs/#/坚果云)

- [手动同步](http://ksria.com/simpread/docs/#/手动同步)

- [自动同步](http://ksria.com/simpread/docs/#/自动同步)

- [同步助手](http://ksria.com/simpread/docs/#/Sync)

- [高级账户](http://ksria.com/simpread/docs/#/高级账户)

**Comment by xiang578** [comment]
开放平台支持使用 dropbox 吗？

**Reply by Kenshin** [reply]
之前是可以的，现在不支持，细节 [看这里](https://github.com/Kenshin/simpread/discussions/3171)
